packages/components/src/DataList/DataList.test.tsx

```diff
@@ -388,14 +396,70 @@ describe("DataList", () => {
     });
   });
 
+  describe("DataListStickyHeader", () => {
+    const layoutWrapper = "layout-wrapper";
+    const layoutItem = "layout-item";
+    const mockOnSearch = jest.fn();
+
+    const renderDataListWithLayout = (children: ReactElement) => {
+      render(
+        <DataList data={mockData} headers={{}}>
+          {children}
+          <DataList.Layout>
+            {(item: DataListItemType<typeof mockData>) => (
+              <div data-testid={layoutWrapper}>
+                <div data-testid={layoutItem}>{item.name}</div>
+                <div data-testid={layoutItem}>{item.email}</div>
+              </div>
+            )}
+          </DataList.Layout>
+        </DataList>,
+      );
+    };
+
+    it("should render the default layout and Sticky Header if DataList.Search is provided", () => {
+      renderDataListWithLayout(<DataList.Search onSearch={mockOnSearch} />);
+      expect(
+        screen.queryByTestId(DATA_LIST_STICKY_HEADER_TEST_ID),
+      ).toBeInTheDocument();
+      expect(screen.getAllByTestId(layoutWrapper)).toHaveLength(2);
+      expect(screen.getAllByTestId(layoutItem)).toHaveLength(4);
+    });
+
+    it("should render the default layout and Sticky Header if DataList.Filters is provided", () => {
+      renderDataListWithLayout(
+        <DataList.Filters>
+          <div>Filters</div>
+        </DataList.Filters>,
+      );
+      expect(
+        screen.queryByTestId(DATA_LIST_STICKY_HEADER_TEST_ID),
+      ).toBeInTheDocument();
+      expect(screen.getAllByTestId(layoutWrapper)).toHaveLength(2);
+      expect(screen.getAllByTestId(layoutItem)).toHaveLength(4);
+    });
+
+    it("should render the default layout without Sticky Header if DataList.Search and DataList.Filters are not provided", () => {
+      renderDataListWithLayout(<></>);
+      expect(
+        screen.queryByTestId(DATA_LIST_STICKY_HEADER_TEST_ID),
+      ).not.toBeInTheDocument();
+      expect(screen.getAllByTestId(layoutWrapper)).toHaveLength(2);
```

Comment:
You're right!  We don't need those extra assertions since they aren't related to the sticky header being present. I've made changes to the test here 3c2f7a8e608a840ddab06b03f6d7127274740228